rir/src/utils/FunctionWriter.h

```diff
@@ -91,39 +61,35 @@ class FunctionHandle {
 
             TYPEOF(newStore) = EXTERNALSXP;
 
-            memcpy(newPayload, payload, capacity);
-            memset(payload, 0xee, capacity);
+            memcpy(newPayload, function, capacity);
 
             R_PreserveObject(newStore);
-            R_ReleaseObject(store);
+            R_ReleaseObject(function->container());
 
-            assert(function == payload);
-            store = newStore;
-            payload = newPayload;
-            function = (Function*)payload;
+            function = reinterpret_cast<Function*>(newPayload);
             capacity = newCapacity;
         }
 
         unsigned offset = function->size;
-        void* insert = (void*)((uintptr_t)payload + function->size);
+        void* insert = (void*)((uintptr_t)function + function->size);
```

Comment:
`reinterpret_cast<void*>(reinterpret_cast<uintptr_t>(function) + function->size)`?
(gosh that's ugly)